### PR TITLE
feat: ✨ architecture rewrite and breaking changes

### LINKED ISSUE #460: Dependency Dashboard
This issue lists Renovate updates and detected dependencies. Read the [Dependency Dashboard](https://docs.renovatebot.com/key-concepts/dashboard/) docs to learn more.<br>[View this repository on the Mend.io Web Portal](https://developer.mend.io/github/solid-design-system/solid).

## Pending Approval

The following branches are pending approval. To create them, click on a checkbox below.

 - [ ] chore(deps): update dependency @types/node to v26.6.2
 - [ ] chore(deps): update dependency chromatic to v18.9.4
 - [ ] chore(deps): update dependency cssnano to v9.0.5
 - [ ] chore(deps): update dependency jest to v30.5.2
 - [ ] chore(deps): update dependency prettier to v3.9.8
 - [ ] chore(deps): update dependency style-dictionary to v5.5.4
 - [ ] chore(deps): update tailwindcss monorepo to v4.3.3 (`@tailwindcss/node`, `@tailwindcss/oxide`, `@tailwindcss/postcss`, `@tailwindcss/vite`, `tailwindcss`)
 - [ ] chore(deps): update dependency eslint to v10.11.0
 - [ ] chore(deps): update dependency node to v24.21.0
 - [ ] chore(deps): update dependency typescript to v5.9.3
 - [ ] chore(deps): update dependency typescript to v5.9.3
 - [ ] chore(deps): update actions/checkout action to v7
 - [ ] chore(deps): update actions/setup-node action to v7
 - [ ] chore(deps): update amannn/action-semantic-pull-request action to v6
 - [ ] chore(deps): update dependency dotenv to v18
 - [ ] chore(deps): update dependency eslint to v10
 - [ ] chore(deps): update dependency typescript to v7
 - [ ] chore(deps): update dependency wc-storybook-helpers to v2
 - [ ] chore(deps): update louisbrunner/checks-action action to v3
 - [ ] chore(deps): update peter-evans/find-comment action to v4
 - [ ] chore(deps): update pnpm/action-setup action to v6
 - [ ] 🔐 **Create all pending approval PRs at once** 🔐

## Pending Status Checks

The following updates await pending status checks. To force their creation now, click on a checkbox below.

 - [ ] chore(deps): update dependency @modelcontextprotocol/sdk to v1.30.1
 - [ ] chore(deps): update dependency ts-jest to v29.4.13
 - [ ] chore(deps): update dependency tsx to v4.23.15
 - [ ] chore(deps): update typescript-eslint monorepo to v8.70.1 (`@typescript-eslint/eslint-plugin`, `@typescript-eslint/parser`, `@typescript-eslint/rule-tester`, `@typescript-eslint/utils`)

## Open

The following updates have all been created. To force a retry/rebase of any, click on a checkbox below.

 - [ ] [chore(deps): update dependency vite to v8.0.16 [security]](../pull/3274)
 - [ ] [chore(deps): update pnpm to v10.34.5 [security]](../pull/3244)
 - [ ] **Click on this checkbox to rebase all open PRs at once**

## Detected Dependencies

<details><summary>github-actions (8)</summary>
<blockquote>

<details><summary>.github/workflows/check-pr.yml (10)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `chromaui/action latest`
 - `mcr.microsoft.com/playwright v1.63.0-noble`
 - `node 24.14.1` → [Updates: `24.21.0`]
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

<details><summary>.github/workflows/chromatic-main.yml (5)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `chromaui/action latest`
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

<details><summary>.github/workflows/chromatic-next.yml (5)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `chromaui/action latest`
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

<details><summary>.github/workflows/lint-pr.yml (1)</summary>

 - `amannn/action-semantic-pull-request v5` → [Updates: `v6`]

</details>

<details><summary>.github/workflows/release.yml (8)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `node 24.14.1` → [Updates: `24.21.0`]
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

<details><summary>.github/workflows/size-check.yml (6)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `LouisBrunner/checks-action v2.0.0` → [Updates: `v3.1.0`]
 - `LouisBrunner/checks-action v2.0.0` → [Updates: `v3.1.0`]
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

<details><summary>.github/workflows/sync-to-azure.yml (1)</summary>

 - `actions/checkout v6` → [Updates: `v7`]

</details>

<details><summary>.github/workflows/upload-storybook-to-github-pages.yml (8)</summary>

 - `actions/checkout v6` → [Updates: `v7`]
 - `actions/setup-node v6` → [Updates: `v7`]
 - `pnpm/action-setup v5` → [Updates: `v6`]
 - `actions/upload-pages-artifact v5`
 - `actions/deploy-pages v5`
 - `peter-evans/find-comment v3` → [Updates: `v4`]
 - `peter-evans/create-or-update-comment v5`
 - `node 24.14.1` → [Updates: `24.21.0`]

</details>

</blockquote>
</details>

<details><summary>npm (11)</summary>
<blockquote>

<details><summary>package.json (25)</summary>

 - `@changesets/cli ^3.0.3`
 - `@eslint/compat ^2.1.1`
 - `@eslint/eslintrc ^3.3.7`
 - `@eslint/js ^10.0.1`
 - `@octokit/rest ^22.0.1`
 - `@typescript-eslint/eslint-plugin ^8.70.0` → [Updates: `^8.70.0`]
 - `@typescript-eslint/parser ^8.70.0` → [Updates: `^8.70.0`]
 - `changesets-changelog-clean ^1.4.0`
 - `eslint ^10.10.0` → [Updates: `^10.10.0`]
 - `eslint-config-prettier ^10.1.8`
 - `eslint-plugin-chai-expect ^4.1.0`
 - `eslint-plugin-chai-friendly ^1.2.1`
 - `eslint-plugin-import ^2.32.0`
 - `eslint-plugin-lit ^2.3.1`
 - `eslint-plugin-lit-a11y ^5.1.1`
 - `eslint-plugin-storybook ^10.6.0`
 - `eslint-plugin-wc ^3.1.0`
 - `globals ^17.12.0`
 - `globby ^16.2.4`
 - `husky ^9.1.7`
 - `lint-staged ^17.5.1`
 - `prettier ^3.9.7` → [Updates: `^3.9.7`]
 - `node ^24.14.1`
 - `pnpm ^10.28.2` → [Updates: `^10.34.5`]
 - `pnpm 10.28.2+sha512.41872f037ad22f7348e3b1debbaf7e867cfd448f2726d9cf74c08f19507c31d2c8e7a11525b983febc2df640b5438dee6023ebb1f84ed43cc2d654d2bc326264` → [Updates: `10.34.5`]

</details>

<details><summary>packages/components/package.json (51)</summary>

 - `@floating-ui/dom ^1.8.0`
 - `@shoelace-style/animations ^1.2.0`
 - `@shoelace-style/localize ^3.2.3`
 - `classix ^2.2.19`
 - `lit ^3.3.3`
 - `style-observer ^0.1.2`
 - `@custom-elements-manifest/analyzer ^0.11.0`
 - `@open-wc/testing ^5.0.0`
 - `@open-wc/testing-helpers ^3.0.1`
 - `@tailwindcss/node ^4.3.2` → [Updates: `^4.3.2`]
 - `@tailwindcss/oxide ^4.3.2` → [Updates: `^4.3.2`]
 - `@tailwindcss/postcss ^4.3.2` → [Updates: `^4.3.2`]
 - `@tarekraafat/autocomplete.js ^10.2.10`
 - `@types/mocha ^10.0.10`
 - `@types/sinon ^22.0.0`
 - `@web/dev-server-esbuild ^2.0.0`
 - `@web/test-runner ^1.0.0`
 - `@web/test-runner-commands ^1.0.1`
 - `@web/test-runner-playwright ^1.0.0`
 - `autoprefixer ^10.6.1`
 - `change-case ^5.4.4`
 - `chokidar ^5.0.0`
 - `command-line-args ^6.0.2`
 - `comment-parser ^1.4.9`
 - `concurrently ^10.0.5`
 - `cssnano ^9.0.4` → [Updates: `^9.0.4`]
 - `custom-element-jet-brains-integration ^1.8.0`
 - `custom-element-vs-code-integration ^1.5.0`
 - `del ^8.0.1`
 - `esbuild ^0.28.2`
 - `esbuild-plugin-replace ^2.0.0`
 - `esbuild-plugin-tailwindcss ^2.2.0`
 - `globby ^16.2.4`
 - `gzip-size ^7.0.0`
 - `jsonata ^2.2.2`
 - `lightningcss ^1.33.0`
 - `lit-html ^3.3.3`
 - `minify-html-literals ^1.3.5`
 - `ora ^9.4.1`
 - `picocolors ^1.1.1`
 - `playwright ^1.63.0`
 - `plop ^4.0.5`
 - `postcss ^8.5.28`
 - `postcss-import ^17.0.0`
 - `postcss-nested ^8.0.1`
 - `prettier ^3.9.7` → [Updates: `^3.9.7`]
 - `recursive-copy ^2.0.14`
 - `sinon ^22.1.0`
 - `tailwindcss ^4.3.2` → [Updates: `^4.3.2`]
 - `typescript 5.8.3` → [Updates: `5.9.3`, `7.0.2`]
 - `user-agent-data-types ^0.4.3`

</details>

<details><summary>packages/docs/package.json (48)</summary>

 - `@storybook/addon-a11y ^10.6.0`
 - `classix ^2.2.19`
 - `custom-elements-manifest ^2.1.0`
 - `lit ^3.3.3`
 - `vite-plugin-replace ^0.1.1`
 - `@open-wc/testing-helpers ^3.0.1`
 - `@playwright/test ^1.63.0`
 - `@rollup/pluginutils ^5.4.0`
 - `@storybook/addon-docs ^10.6.0`
 - `@storybook/addon-links ^10.6.0`
 - `@storybook/addon-themes ^10.6.0`
 - `@storybook/test-runner ^0.24.5`
 - `@storybook/web-components-vite ^10.6.0`
 - `@tailwindcss/postcss ^4.3.2` → [Updates: `^4.3.2`]
 - `@tailwindcss/vite ^4.3.2` → [Updates: `^4.3.2`]
 - `@tarekraafat/autocomplete.js ^10.2.10`
 - `@types/node ^26.6.1` → [Updates: `^26.6.1`]
 - `autoprefixer ^10.6.1`
 - `axe-html-reporter ^2.2.11`
 - `axe-playwright ^2.2.2`
 - `change-case ^5.4.4`
 - `chromatic ^18.9.0` → [Updates: `^18.9.0`]
 - `comment-parser ^1.4.9`
 - `concurrently ^10.0.5`
 - `countup.js ^2.10.1`
 - `cssnano ^9.0.4` → [Updates: `^9.0.4`]
 - `esbuild ^0.28.2`
 - `globby ^16.2.4`
 - `http-server ^14.1.1`
 - `lit-html ^3.3.3`
 - `normalize.css ^8.0.1`
 - `picocolors ^1.1.1`
 - `postcss ^8.5.28`
 - `postcss-import ^17.0.0`
 - `postcss-nested ^8.0.1`
 - `prettier ^3.9.7` → [Updates: `^3.9.7`]
 - `react ^19.3.0`
 - `react-dom ^19.3.0`
 - `storybook ^10.6.0`
 - `tailwindcss ^4.3.2` → [Updates: `^4.3.2`]
 - `typescript 5.9.3` → [Updates: `7.0.2`]
 - `user-agent-data-types ^0.4.3`
 - `vite 8.0.8` → [Updates: `8.0.16`]
 - `vite-node ^6.0.0`
 - `vite-plugin-cem ^0.8.5`
 - `vite-tsconfig-paths ^6.1.1`
 - `wait-on ^9.1.0`
 - `wc-storybook-helpers 1.5.3` → [Updates: `2.0.4`]

</details>

<details><summary>packages/eslint-plugin/package.json (15)</summary>

 - `@html-eslint/parser ^0.66.0`
 - `@types/node ^26.6.1` → [Updates: `^26.6.1`]
 - `@typescript-eslint/rule-tester ^8.70.0` → [Updates: `^8.70.0`]
 - `@typescript-eslint/utils ^8.70.0` → [Updates: `^8.70.0`]
 - `es-html-parser ^0.3.2`
 - `esbuild ^0.28.2`
 - `eslint ^10.10.0` → [Updates: `^10.10.0`]
 - `globby ^16.2.4`
 - `jest ^30.5.1` → [Updates: `^30.5.1`]
 - `plop ^4.0.5`
 - `ts-jest ^29.4.12` → [Updates: `^29.4.12`]
 - `ts-node ^10.9.2`
 - `typescript 5.9.3` → [Updates: `7.0.2`]
 - `@html-eslint/parser ^0.66.0`
 - `eslint ^9.18.0` → [Updates: `^9.18.0 || ^10.0.0`]

</details>

<details><summary>packages/mcp/package.json (11)</summary>

 - `@modelcontextprotocol/sdk ^1.30.0` → [Updates: `^1.30.0`]
 - `globby ^16.2.4`
 - `zod ^4.6.5`
 - `@types/node ^26.6.1` → [Updates: `^26.6.1`]
 - `change-case ^5.4.4`
 - `custom-elements-manifest ^2.1.0`
 - `eslint ^10.10.0` → [Updates: `^10.10.0`]
 - `ora ^9.4.1`
 - `prettier ^3.9.7` → [Updates: `^3.9.7`]
 - `tsx ^4.23.13` → [Updates: `^4.23.13`]
 - `typescript ^5.9.3` → [Updates: `^7.0.0`]

</details>

<details><summary>packages/placeholders/package.json (1)</summary>

 - `typescript 5.9.3` → [Updates: `5.9.3`, `7.0.2`]

</details>

<details><summary>packages/styles/package.json (14)</summary>

 - `@tailwindcss/postcss ^4.3.2` → [Updates: `^4.3.2`]
 - `autoprefixer ^10.6.1`
 - `command-line-args ^6.0.2`
 - `cssnano ^9.0.4` → [Updates: `^9.0.4`]
 - `del ^8.0.1`
 - `globby ^16.2.4`
 - `ora ^9.4.1`
 - `picocolors ^1.1.1`
 - `plop ^4.0.5`
 - `postcss ^8.5.28`
 - `postcss-cli ^12.0.0`
 - `postcss-nested ^8.0.1`
 - `prettier ^3.9.7` → [Updates: `^3.9.7`]
 - `tailwindcss ^4.3.2` → [Updates: `^4.3.2`]

</details>

<details><summary>packages/theming/package.json (2)</summary>

 - `chroma-js ^3.1.2`
 - `typescript 5.9.3` → [Updates: `7.0.2`]

</details>

<details><summary>packages/tokens/package.json (10)</summary>

 - `@figma-export/types ^6.4.1`
 - `@figma/rest-api-spec ^0.42.0`
 - `@tamtamchik/json-deep-sort ^1.5.0`
 - `@tokens-studio/sd-transforms ^2.0.3`
 - `dotenv ^17.4.2` → [Updates: `^18.0.0`]
 - `ora ^9.4.1`
 - `picocolors ^1.1.1`
 - `style-dictionary ^5.5.3` → [Updates: `^5.5.3`]
 - `tailwindcss ^4.3.2` → [Updates: `^4.3.2`]
 - `tailwindcss-export-config ^4.1.0`

</details>

<details><summary>packages/versioning/package.json</summary>

</details>

<details><summary>pnpm-workspace.yaml</summary>

</details>

</blockquote>
</details>

---

- [ ] Check this box to trigger a request for Renovate to run again on this repository

### LINKED ISSUE #859: fix: loading a single component loads all components
## Current behavior
Loading a single component via CDN triggers loading of ALL components

## Expected behavior
Only the requested component is loaded

## Steps to reproduce
Save as .html and run in browser:
```
<!DOCTYPE html>
<html>
  <head>
  </head>
  <body>
    <script src="https://solid-design-system.fe.union-investment.de/2.8.2/versioned-components/es/button.js" type="module"></script>
  </body>
</html>
```

## Technical Information
Loading button.js works as expected.
However, it imports and loads ./solid-components2.js which in turn imports and loads all other components.

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1112: ci: 👷 Random Playwright Fails due to Import error (Chromium)
## User Story
As a Developer for the Solid Design System, I would like to remove, or at least, understand why our CI piplines randomly fails due to a "Could not import test module" error that is only resolved with subsequent retries, so that I can push error free PRs and move on to my next task without worrying about random/wrong errors that slow down development in general. 

## Technical Information
Below is the logs for the failing CI build specifically the *install, verify and build check*. The specific area of interest for me was this line:  "❌ Could not import your test module. Check the browser logs or open the browser in debug mode for more information.  (failed on Chromium)"

[0_Install, verify and build.txt](https://github.com/solid-design-system/solid/files/15402827/0_Install.verify.and.build.txt)

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1310: fix: 🤔 Default selection for sd-flipcard
## Current behavior
Currently, the default variant for Flipcard in the code is the `empty` variant. If a user selects and uses an image, low contrast may occur, potentially creating an accessibility issue.

## Expected behavior
Set a different default value to prevent using the component with an image background without considering the contrast. In Figma, the default variant is `primary`, which would be a good approach to implement in the code as well.

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1339: feat[dev]: ✨ brandshape image without default slot
## User Story
As a user of the Solid Design System, I want to display images with the brandshape and there should be no option to include content on top of the image. 

### Suggested Solution
If the chosen variant is `image`, the default slot should not be rendered.

## DoR
- [x] Item has business value
- [x] Item has been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1375: feat: ✨ BREAKING CHANGES
## User Story
Epic issue to collect all breaking changes.

All subtasks should be merged at once (maybe even together with the rewrite docs PR).

## Subtasks
- [x] #1296 
- [x] #1339 
- [x] #1310 
- [x] #1314
- [x] #1418

## DoR
- [x] Item has business value
- [ ] Item all subtasks have been estimated by the team
- [x] Item is clear and well-defined
- [x] Item dependencies have been identified

## DoD
- [ ] All subtasks have been closed

### LINKED ISSUE #1430: feat[dev]: ✨ implement reverse layout logic for front and back
## Description
This tasks describes the Figma implementation for: #1314 

While doing this task please make sure you update the template using the bottom variants. Done in figma.

### Open Questions towards design
- [ ] Question1
- [ ] Question2

## DoR
- [ ] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1544: feat[dev]: ✨ implement new sd-flipcard
## Description
This tasks describes the Figma implementation for: #1540

1. Removing the activation property
2. adding an icon

### Open Questions towards design
- [ ] Question1
- [ ] Question2

## DoR
- [x] Item has been estimated by the team
- [x] Item dependencies have been identified and documented

## DoD
- [ ] Documentation has been created/updated (if applicable)
- [ ] Migration Guide has been created/updated (if applicable)
- [ ] Relevant E2E tests (Features, A11y, Bug fixes) are created/updated
- [ ] Relevant stories (Features, A11y) are created/updated
- [ ] Implementation works successfully on `feature` branch

### LINKED ISSUE #1570: feat: ✨ refactor styles into seperate package
(This is a spin off of https://github.com/solid-design-system/solid/issues/1131)

## User Story
As a maintainer of the Solid Design System, I would like to have a more scalable and robust infrastructure, so that we're prepared for upcoming features and to better understand how parts of the design system interact with each other.

### Suggested Solution

1. We move `/components/src/styles` into a separate package `/styles` like we did in [Synergy (Styles PR)](https://github.com/synergy-design-system/synergy-design-system/pull/495). This will need PostCSS in styles package, including TailwindCSS, nesting etc. Ideally this should create single modules of the CSS like we do in Synergy as well. This will need separate `package.json` etc.

## Technical Information

There will be tough nuts to crack, like importing `styles` into `components` including live updates, rewriting the build scripts etc.

One important part will be improving and fixing the builds. This might become a really big change. Currently the final build looks like the following:

```
dist/
     components/
                es/
                   solid-components.js
                   solid-components2.js (containing utilities)
                   input.js
                   umd/solid-components.js
       package/
               components/
                          solid-components.js
                          utilities/
                                    autocomplete-config.js
                          components/
                                     input.js
       components/
                  es/
                     solid-components.js
                     solid-components2.js (containing utilities)
                     input.js
                  umd/solid-components.js
```

There are big problems coming with the current builds, e. g. that you can't select `utilties` in an appropriate file from the `components` folder.

I think we should simplify everything here:

```
dist/
     components/
                solid-components.js
                solid-components.min.js
                solid-components.umd.js (containing the whole bundle!)
                solid-components.umd.min.js (containing the whole minified bundle!)
                utilities/
                          autocomplete-config.js
                          autocomplete-config.min.js
                          components/
                                     input.js
                                     input.min.js
       versioned-components/
                         ...
```

## Subtasks
- [ ] Design_implementation_task
- [ ] Development_implementation_task
- [ ] Design_release_task

## DoR
- [ ] Item has business value
- [ ] Item all subtasks have been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] All subtasks have been closed

### LINKED ISSUE #1571: feat: ✨ refactor build in components
(This is a spin off of https://github.com/solid-design-system/solid/issues/1131)

## User Story
As a maintainer of the Solid Design System, I would like to have a more scalable and robust infrastructure, so that we're prepared for upcoming features and to better understand how parts of the design system interact with each other.

### Suggested Solution

1. The build scripts in `/components` focus only the build step, similar to what [Shoelace (build.js)](https://github.com/shoelace-style/shoelace/blob/next/scripts/build.js) and [Synergy (build.js)](https://github.com/synergy-design-system/synergy-design-system/blob/main/packages/components/scripts/build.js) does. Our `cdn` and `package` separation should be refactored as written below. Furthermore this should be fixed during that: https://github.com/solid-design-system/solid/issues/859#issuecomment-2017457770
2. Tests use these build scripts instead of the current Vite setup. I expect that this will solve #1112 as well.

## Technical Information

There will be tough nuts to crack, like importing `styles` into `components` including live updates, rewriting the build scripts etc.
The biggest challenge will be to make the vite magic (being used in Storybook) fully compatible with `esbuild` and making the tests work at the same time.

One important part will be improving and fixing the builds. This might become a really big change. Currently the final build looks like the following:

```
dist/
     components/
                es/
                   solid-components.js
                   solid-components2.js (containing utilities)
                   input.js
                   umd/solid-components.js
       package/
               components/
                          solid-components.js
                          utilities/
                                    autocomplete-config.js
                          components/
                                     input.js
       versioned-components/
                  es/
                     solid-components.js
                     solid-components2.js (containing utilities)
                     input.js
                  umd/solid-components.js
```

There are big problems coming with the current builds, e. g. that you can't select `utilties` in an appropriate file from the `components` folder.

I think we should aim for the following, whcih works well for Shoelace:

```
dist/
       solid-components.js
       solid-components.iiff.js (containing the whole bundle!)
                utilities/
                          autocomplete-config.js
                          autocomplete-config.min.js
                          components/
                                     input.js
                                     input.min.js
versioned-dist/
                         ...
cdn/
       (everything minified including bundled lit etc)
       solid-components.js (containing the whole bundle!)
       solid-components.iiff.js (containing the whole bundle!)
                utilities/
                          autocomplete-config.js
                          autocomplete-config.min.js
                          components/
                                     input.js
                                     input.min.js
versioned-cdn/
                         ...
```

## Subtasks
- [x] Make tests running without any fails @paulovareiro29 – checkout how Synergy and Shoelace did it, including some more specific tests like the one for the `sd-icon` 
- [x] extract regex transformation from `esbuild-plugin-lit-tailwind.js` into own function that is exported; move `vite-plugin-lit-tailwind.ts` into `packages/docs` and exchange the transformation with the exported function from `esbuild-plugin-lit-tailwind` @paulovareiro29 
- [x] harmonize import of global styles following Lit's proposal and get rid of component.styles and multiple imports @mariohamann 
- [x] Setup Meta data
- [x] Setup versioning for components
- [x] Setup versioning for meta data (append data vs. new file?)
- [x] Setup versioning for styles
- [x] make styles structure flat,, only single modules, prose and index
- [x] make styles work with Storybook
- [x] Export and publish `_utilties.css`, as it has to be imported in the very end
- [x] Make linters etc. work again
- [ ] Update docs and make section about breaking changes (?)
- [ ] Update installation guides
- [ ] Setup Releases for Styles package
- [x] re-add cssnano

## DoR
- [ ] Item has business value
- [ ] Item all subtasks have been estimated by the team
- [ ] Item is clear and well-defined
- [ ] Item dependencies have been identified

## DoD
- [ ] All subtasks have been closed

### PR DESCRIPTION
**Pull request by mariohamann**
## Description:
This PR is a complete rebuild of how components and styles are structured and built.

1. `components` are now using `esbuild` for faster, more predicatable and more influenceable builds (#1571). Cherry-picking is now possible through this change as well (fix #859). Furthermore this will enable us to make it possible to create React wrappers, Vue types, typed events etc. in the near future. Tests were rewritten so that they work again (fix #1112).
2. `styles` are extracted in an own package (#1570), so that they can be versioned and built independentely. Furthermore the needed CSS which was needed for the components is now separeted from styles. Cherry-picking is now possible as well. The switch furthermore enabled us to switch to JSDocs-like docs to create Storybook knobs.
3. CDN URLs will be rewritten and wildcard URLs won't be provisioned anymore. This is a preparation for #1651.
4. Documentation regarding installation and contribution is and will be separated as well, so that this more connected to the actual package in future.
5. In general lots of code was refactored and touched to simplify things. Old Shoelace stuff, that was not needed anymore was removed. This is an ongoing process and is not finally done.
6. In addition #460 + #1400 will be closed as most or all packages are even being updated.
7. `CONTRIBUTION` guidelines were now separated (from `DEVELOPER_STARTING_GUIDE`) to be more focused on the specific package. Furthermore they were added to Storybook. `pnpm plop` was updated for components and created for `styles`.
8. Migration guides for Breaking Changes were introduced and added to Storybook.

## TBD
- [x] `.iief` build
- [x] Contribution guidelines / Plop
- [x] fix Storybook controls (or follow-up ticket)
- [x] Breaking Changes guidelines
- [x] codepen
- [x] integrate other breaking changes from  #1429
- [x] prepare styles package to be published
- [x] add changesets

---

- close #1544
- close #1430
- close #1310
- close #1339 
- close #1112 
- close #859 
- close #460 
- close #1375 
- close #1570 
- close #1571 

## Definition of Reviewable:
- [x] Documentation is created/updated
- [x] Migration Guide is created/updated
- [x] E2E tests (features, a11y, bug fixes) are created/updated
<!-- *If this PR includes a bug fix, an E2E test is necessary to verify the change. If the fix is purely visual, ensuring it is captured within our chromatic screenshot tests is sufficient.* -→
- [ ] Stories (features, a11y) are created/updated
- [ ] relevant tickets are linked

### CODE REVIEW

File: packages/components/scripts/build.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+await nextTask('Versioning components and meta data', async () => {
+  await execPromise('node scripts/make-versioning.js', { stdio: 'inherit' });
+});
+
+// let result;
+
+// // Log deferred output
+// if (result.output.length > 0) {
+//   console.log('\n' + result.output.join('\n'));
+// }
```
Comment (Vahid1919) [inline packages/components/scripts/build.js:153]:
```suggestion
```

found some commented out code. should this removed?

**Review by Vahid1919 — APPROVED** [review]
left one comment, rest LGTM!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -30,10 +30,24 @@
...
+
+- Solid Components follows a monorepo structure with packages (e. g. `components`) managed by `pnpm` (which is a replacement for `npm`). Linting and Formatting is centralized at root level. Packages have to be run individually (e.g. `cd packages/docs && pnpm dev` to start development server). Run `pnpm verify` at the root directory periodically, particularly, before pushing changes when a pull request is already opened.
+
+- While the following guidelines give you a deeper understanding of the Solid Design system as a whole, each package has its own `CONTRIBUTING.md` file, which provides detailed information on how to contribute to the respective package. Ensure to follow the guidelines outlined in these files when contributing to a specific package.
+  - [Components](./?path=/docs/packages-components-contributing--docs)
+  - [Styles](./?path=/docs/packages-styles-contributing--docs)
+  - [Placeholders](./?path=/docs/packages-placeholders-contributing--docs)
+  - [Docs](./?path=/docs/packages-docs-contributing--docs)
```
[the remark this answers is not in the document]
Comment (mariohamann) [reply CONTRIBUTING.md:50]: Not really. It was more a priority thing to not overexceed this PR: We already had issues on the placeholder package with contributions, that's why I added them. I now added one for `tokens` as well (based on existing information). For `theming` would be nice, but has low prio I believe.

File: packages/components/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+If you are working on a component, follow the steps below:
+
+1. run `cd packages/components && pnpm plop` to generate a new component. This creates:
+   1. A new TS file: `./packages/components/src/components/*/*.ts`
+   2. A new Test file: `./packages/components/src/components/*/*.test.ts`
+   3. A new documentation story: `./packages/docs/src/stories/components/*/*.stories.ts`
+   4. A new test story: `./packages/docs/src/stories/components/*/*.test.stories.ts`
+   5. An update to `./packages/components/src/solid-components.ts` to include the new component.
+
+**Run `pnpm verify` in the root directory before creating a PR.** This will check that all formatting, tests, and build processes are working correctly to allow the pipeline to run successfully.
```
[the remark this answers is not in the document]
Comment (mariohamann) [reply packages/components/CONTRIBUTING.md:14]: Changed and added it in Styles as well.

File: packages/components/src/components/teaser-media/teaser-media.ts [hunk 4]
```diff
@@ -125,6 +125,14 @@ export default class SdTeaserMedia extends SolidElement {
       ::slotted([slot='headline']) {
         @apply font-bold !m-0 !text-lg;
       }
+
+      @media (min-width: 768px) {
+        part[expandable] {
```
[the remark this answers is not in the document]
Comment (mariohamann) [reply packages/components/src/components/teaser-media/teaser-media.ts:130]: I did undo my changes. Thanks for the heads-up.

File: packages/components/src/components/teaser-media/teaser-media.ts [hunk 5]
```diff
@@ -29,7 +29,7 @@ import SolidElement from '../../internal/solid-element';
 @customElement('sd-teaser-media')
 export default class SdTeaserMedia extends SolidElement {
   @property({ reflect: true })
-  variant: 'white' | 'neutral-100' | 'primary' | 'primary-100' | 'gradient-white' | 'gradient-dark' = 'white';
+  variant: 'white' | 'neutral-100' | 'primary' | 'primary-100' | 'gradient-light' | 'gradient-dark' = 'white';
```
[the remark this answers is not in the document]
Comment (mariohamann) [reply packages/components/src/components/teaser-media/teaser-media.ts:32]: I switched to the current version from main. I believe this was a merge error.

**Review by smfonseca — APPROVED** [review]

**Pull request merged**